components/custom_board/pcm51xx/pcm51xx.c

```diff
@@ -138,44 +164,76 @@ vTaskDelay(200 / portTICK_PERIOD_MS);
 }
 
 esp_err_t pcm51xx_set_volume(int vol) {
-  // vol is given as 1/2dB step with
-  // 255: -inf (mute)
-  // 254: -103dB
-  // 48:  0dB
-  // 0 (max): +24dB
-
-  if (vol < PCM51XX_VOLUME_MIN) {
-    vol = PCM51XX_VOLUME_MIN;
+  ESP_LOGD(TAG, "%s: vol=%d%%", __func__, vol);
+  // vol is given as percentage (0-100%)
+  // Input range: 0% (min) to 100% (max/0dB)
+  // 
+  // PCM51XX register mapping (1/2dB steps):
+  // 0x30 (48 decimal):  0dB    <- 100%
+  // 0xFF (255):        -inf (mute) <- 0%
+  //
+  // Inverted mapping: higher percentage = higher register value (less attenuation)
+
+  // Clamp input to valid range
+  if (vol < 0) {
+    vol = 0;
+  }
+  if (vol > 100) {
+    vol = 100;
   }
-  if (vol > PCM51XX_VOLUME_MAX) {
-    vol = PCM51XX_VOLUME_MAX;
+
+  uint8_t register_value;
+  
+  if (vol == 0) {
+    // 0% = mute
+    register_value = 0xFF;
+  } else {
+    // Map 1-100% to register values 0xFE-0x30
+    // Linear mapping: 100% -> 0x30 (48), 1% -> 0xFE (254)
+    // Formula: reg_val = 0x30 + (100 - vol) * (0xFE - 0x30) / 99
+    //        = 48 + (100 - vol) * 206 / 99
+    register_value = 0x30 + ((100 - vol) * 206) / 99;
   }
+
   uint8_t cmd[2] = {0, 0};
   esp_err_t ret = ESP_OK;
 
-  cmd[1] = vol;
+  cmd[1] = register_value;
 
   cmd[0] = PCM51XX_REG_VOL_L;
   ret = i2c_bus_write_bytes(i2c_handler, pcm51xx_addr, &cmd[0], 1, &cmd[1], 1);
```

Comment:
Generally: seems to be well done.

But: I saw in some [recent conversations in current issues](https://github.com/CarlosDerSeher/snapclient/issues/168#issuecomment-3575058311), the log complains about the usage of an old i2c driver and a migration to i2c_master is encouraged. I don't request you do this now but maybe we should keep that in mind and find the time to do it at some point, at least we should add new DAC with the new i2c idf implementation